src/game.cpp

```diff
@@ -1113,24 +1115,31 @@ static inline void create_formspec_menu(GUIFormSpecMenu **cur_formspec,
 		JoystickController *joystick,
 		IFormSource *fs_src, TextDest *txt_dest, Client *client)
 {
-
-	if (*cur_formspec == 0) {
-		*cur_formspec = new GUIFormSpecMenu(device, joystick,
-			guiroot, -1, &g_menumgr, invmgr, gamedef, tsrc,
-			fs_src, txt_dest, client);
-		(*cur_formspec)->doPause = false;
-
-		/*
-			Caution: do not call (*cur_formspec)->drop() here --
-			the reference might outlive the menu, so we will
-			periodically check if *cur_formspec is the only
-			remaining reference (i.e. the menu was removed)
-			and delete it in that case.
-		*/
-
+	verbosestream << "Init GuiFormSpec:" << fs_src->getForm() << std::endl;
+	if (fs_src->getForm() == FORMSPEC_VERSION_STRING && *cur_formspec != 0) {
```

Comment:
As mentioned, I'd like it to ignore the empty string if the source (formname/nodepos) is not the same. If this is from a formname of an empty string, then this check can be avoided